data-conversion/src/main.rs

```diff
@@ -206,21 +226,15 @@ fn binary_to_float(
 fn fixed_to_binary(
     fixed_string: &str,
     filepath_send: &mut File,
+    exp_str: &str,
     // scale: usize,
 ) -> io::Result<()> {
-    let fixed_str;
-    let exp_str;
-    let words: Vec<&str>;
-
-    // Create an array with the elements of fixed_string delinieated by spaces
-    if fixed_string.contains(' ') {
-        // Split the input string into individual words
-        words = fixed_string.split_whitespace().collect();
-        fixed_str = words.first().unwrap_or(&"There is not a fixed number");
-        exp_str = words.get(1).unwrap_or(&"There is no exponent");
-    } else {
-        panic!("Input string does not contain a space.");
-    }
+    // Create an array with the elements of fixed_string
+    let words: Vec<&str> = fixed_string.split_whitespace().collect();
+    let fixed_str: &&str =
+        words.first().unwrap_or(&"There is not a fixed number");
+    // exp_str = words.get(1).unwrap_or(&"There is no exponent");
```

Comment:
I think you might mean `expect()` here, not `unwrap_or()`? `expect()` will panic when there is data missing, which is what I think you want to do. `unwrap_or` will not crash; it will just return this string (and put it in `fixed_str`) when there is no first element.

Reply:
In this case, I think we can simply remove the `Vec` entirely, since now we expect to see only one number per line. (And I think the same can be done in `binary_to_fixed`.)